docs/worldgen/biomemodifier.md

```diff
@@ -0,0 +1,741 @@
+|:------------------------:|:----------------------------------------------------------------------------------------|
+|     `raw_generation`     | First to run. This is used for special terrain-like features such as Small End Islands. |
+|         `lakes`          | Dedicated to spawning pond-like feature such as Lava Lakes.                             |
+|  `local_modifications`   | For modifications to terrain such as Geodes, Icebergs, Boulders, or Dripstone.          |
+| `underground_structures` | Used for small underground structure-like features such as Dungeons or Fossils.         |
+|   `surface_structures`   | For small surface only structure-like features such as Desert Wells.                    |
+|      `strongholds`       | Dedicated for Stronghold structures. No feature is added here in unmodified Minecraft.  |
+|    `underground_ores`    | The step for all Ores and Veins to be added to. This includes Gold, Dirt, Granite, etc. |
+| `underground_decoration` | Used typically for decorating caves. Dripstone Cluster and Sculk Vein are here.         |
+|     `fluid_springs`      | The small Lavafalls and Waterfalls comes from features in this stage.                   |
+|   `vegetal_decoration`   | Nearly all plants (flowers, trees, vines, and more) are added to this stage.            |
+| `top_layer_modification` | Last to run. Used for placing Snow and Ice on the surface of cold biomes.               |
+
+
+## Datagenning Biome Modifiers:
+
+A `BiomeModifier` JSON can be created with [data generation][datagen] by passing a `RegistrySetBuilder` to `DatapackBuiltinEntriesProvider`. The JSON will be placed at `data/<modid>/neoforge/biome_modifier/<path>.json`.
+
+```java
+// Define keys for Datapack registry objects
+
+public static final ResourceKey<BiomeModifier> EXAMPLE_MODIFIER =
+    ResourceKey.create(
+        NeoForgeRegistries.Keys.BIOME_MODIFIERS, // The registry this key is for
+        ResourceLocation.fromNamespaceAndPath(MOD_ID, "example_modifier") // The registry name
+    );
+
+// For some RegistrySetBuilder BUILDER
+// being passed to DatapackBuiltinEntriesProvider
+// in a listener for GatherDataEvent
+BUILDER.add(NeoForgeRegistries.Keys.BIOME_MODIFIERS, bootstrap -> {
+    // Lookup any necessary registries
+    // Static registries only need to be looked up if you need to grab the tag data
+    HolderGetter<Biome> biomes = bootstrap.lookup(Registries.BIOME);
+
+    // Register the Biome Modifiers
+
+    bootstrap.register(EXAMPLE_MODIFIER,
+        new ExampleBiomeModifier(
+            biomes.getOrThrow(Tags.Biomes.IS_OVERWORLD),
+            20
+        )
+    );
+})
+```
+
+```json5
+// In data/examplemod/neoforge/biome_modifier/example_modifier.json
+{
+    // The registy key of the MapCodec for the modifier
+    "type": "examplemod:example_biome_modifier",
+    // All additional settings are applied to the root object
+    "biomes": "#c:is_overworld",
+    "value": 20
+}
```

Comment:
I think this should go after the creating custom modifiers section just because the example used isn't defined until after that section. It also flows a bit more smoothly.